sources/HUBUtilities.h

```diff
@@ -95,6 +116,76 @@ static inline NSDictionary<NSString *, id> * _Nullable HUBMergeDictionaries(NSDi
 }
 
 /**
+ *  Return the property names that should be taken into account when handling a `UINavigationItem`
+ *
+ *  This function is used to determine which properties should be compared or copied. Note that
+ *  `HUBAutoEquatable` is not used here, since we don't control the implementation of the class.
+ */
+static inline NSArray<NSString *> *HUBNavigationItemPropertyNames() {
+    return @[
+        HUBKeyPath((UINavigationItem *)nil, title),
+        HUBKeyPath((UINavigationItem *)nil, titleView),
+        HUBKeyPath((UINavigationItem *)nil, prompt),
+        HUBKeyPath((UINavigationItem *)nil, backBarButtonItem),
+        HUBKeyPath((UINavigationItem *)nil, hidesBackButton),
+        HUBKeyPath((UINavigationItem *)nil, leftBarButtonItems),
+        HUBKeyPath((UINavigationItem *)nil, rightBarButtonItems),
+        HUBKeyPath((UINavigationItem *)nil, leftItemsSupplementBackButton)
+    ];
+}
+
+/**
+ *  Return whether two `UINavigationItem` instances are equal
+ *
+ *  @param navigationItemA The first navigation item
+ *  @param navigationItemB The second navigation item
+ *
+ *  The two instances are considered equal if all properties which have names included in the array
+ *  obtained by calling `HUBNavigationItemPropertyNames()` are equal.
+ */
+static inline BOOL HUBNavigationItemEqualToNavigationItem(UINavigationItem *navigationItemA, UINavigationItem *navigationItemB) {
+    for (NSString * const propertyName in HUBNavigationItemPropertyNames()) {
+        if (!HUBPropertyIsEqual(navigationItemA, navigationItemB, propertyName)) {
+            return NO;
+        }
+    }
+    
+    return YES;
+}
+
+/**
+ *  Copy the properties from an instance of `UINavigationItem` into another
+ *
+ *  @param navigationItemA The navigation item to copy values into
+ *  @param navigationItemB Any navigation item to copy values from
+ *
+ *  @return navigationItemA
+ *
+ *  If `navigationItemB` is nil, all `navigationItemA` properties will be reset to `nil`. To determine
+ *  which properties that should be handled, `HUBNavigationItemPropertyNames()` is called.
+ */
+static inline UINavigationItem *HUBCopyNavigationItemProperties(UINavigationItem *navigationItemA, UINavigationItem * _Nullable navigationItemB) {
```

Comment:
This is so big and complex it feels like it shouldn’t be inlined imo. How about changing it to a normal function?